frontend/src/layouts/EventViewer/EventViewer.tsx

```diff
@@ -84,22 +119,28 @@ export const EventViewer = ({ events, selectedEventCallId, onEventClose, viewerP
 
       <div className="p-4 overflow-auto flex-1">
         <div className="font-mono select-text p-4 rounded-sm bg-muted overflow-x-auto w-full">
-          <ReactJson
-            // Needed to properly handle theme changes
-            key={theme.theme}
-            src={selectedEvent.details}
-            theme={viewerTheme}
-            iconStyle="square"
-            displayDataTypes={false}
-            displayObjectSize={false}
-            indentWidth={4}
-            enableClipboard={true}
-            style={{ fontFamily: "inherit", background: "none" }}
-            shouldCollapse={({ name }) => name !== "root"}
-          />
+          <div ref={jsonViewerRef}>
+            <Marker mark={searchValueMarker.searchValue} >
+              <ReactJson
+                // Needed to properly handle theme changes
+                key={theme.theme}
+                src={selectedEvent.details}
+                theme={viewerTheme}
+                iconStyle="square"
+                displayDataTypes={false}
+                displayObjectSize={false}
+                indentWidth={4}
+                enableClipboard={true}
+                style={{ fontFamily: "inherit", background: "none" }}
+                shouldCollapse={({ name }) => {
+                  return !keysNotToCollapse.flat().includes(name!)
```

Comment:
this one should check the whole key path e.g. `root.arguments[0].user` othewise you'll be opening properties that don't have a match